### PR TITLE
run test pods/containers with CAP_NET_RAW so ping and friends can work

### LINKED ISSUE #133: Pod-to-host ICMP tests fail if NET_RAW capability is not provided by runtime
On a 1.20.4 cluster deployed with CRI-O 1.20.0 and "v6-first" dual-stack, I observe the following failures in the pod-to-host tests (IPs have been altered):

```
---------------------------------------------------------------------------------------------------------------------
🔌 [pod-to-host] Testing cilium-test/client-58dfdc5f6-fkgxg -> 2001:db8:0:1::21...
---------------------------------------------------------------------------------------------------------------------
❌ ping command failed: error in stream: command terminated with exit code 1
❌ [pod-to-host] cilium-test/client-58dfdc5f6-fkgxg (2001:db8:0:0:3::1234) -> 2001:db8:0:1::21 (2001:db8:0:1::21)
```

Exec'ing into the pod shows the failure is caused by insufficient permissions:

```
/ # ping 2001:db8:0:1::21
PING 2001:db8:0:1::21 (2001:db8:0:1::21): 56 data bytes
ping: permission denied (are you root?)
/ # echo $?
1
```

This appears to stem from the pod lacking the `NET_RAW` capability. In my specific case, this happens because CRI-O above 1.18.0 no longer gives `NET_RAW` to containers by default, as implemented in https://github.com/cri-o/cri-o/pull/3119.

However, I assume this would also bite folks who are deploying their clusters with non-CRI-O runtimes but with restrictive pod security contexts lacking `NET_RAW`.

CRI-O offers a few ways to allow ping (enabling `NET_RAW`, enabling the `ip_forward` sysctl, etc) but I'm unsure of the best way to fix this in the more generic context of the Cilium CLI in "all possible container runtimes" (or if it's something that even _should_ be considered here).

Figured it was probably worth opening an issue to discuss before flinging code at the problem.

### PR DESCRIPTION
**Pull request by asteven**
Signed-off-by: Steven Armstrong <steven.armstrong@id.ethz.ch>

Fixes #133 (#270)

### CODE REVIEW

File: connectivity/check/deployment.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ func newDeployment(p deploymentParameters) *appsv1.Deployment {
 							ImagePullPolicy: corev1.PullIfNotPresent,
 							Command:         p.Command,
 							ReadinessProbe:  p.ReadinessProbe,
+                            SecurityContext: &corev1.SecurityContext{
+                                Capabilities: &corev1.Capabilities{
+                                    Add: []corev1.Capability{"NET_RAW"},
+                                },
+                            },
```
Comment (pchaigno) [inline connectivity/check/deployment.go:93]: The indentations and/or closing brackets are somehow messed up.

**Review by pchaigno — CHANGES_REQUESTED** [review]

File: connectivity/check/deployment.go (hunk 1 above)
Comment (asteven) [reply connectivity/check/deployment.go]: Fixed now. Sorry about that. My editor and me usually live in a spaces world ... ;-)

**Review by pchaigno — APPROVED** [review]

**Comment by ti-mo** [comment]
Hi, please add a useful commit message, it currently doesn't provide any context. A short description of the original GH issue would suffice.

Also, please squash into a single commit. Thanks!

**Comment by asteven** [comment]
Better?

**Review by ti-mo — APPROVED** [review]
Perfect, thank you!

**Review by jrajahalme — APPROVED** [review]

**Comment by pchaigno** [comment]
@asteven Could you rebase the pull request on latest master? That should allow the tests to proceed. After that, we should be good to merge :slightly_smiling_face:

**Review by michi-covalent — APPROVED** [review]

**Comment by michi-covalent** [comment]
aks doesn't work from a forked repo. merging 🚀 thank you for the pr @asteven !

**Pull request merged**